spring-ai-core/src/main/java/org/springframework/ai/rag/analysis/query/expansion/QueryExpander.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.rag.analysis.query.expansion;
+
+import org.springframework.ai.rag.Query;
+
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * A component responsible for expanding the input query into a list of related queries
+ * based on a specified strategy. These expansions can be used to capture different
+ * perspectives or to break down complex queries into simpler, more manageable
+ * sub-queries, thereby improving the retrieval process.
+ *
+ * @author Thomas Vitale
+ * @since 1.0.0
+ */
+public interface QueryExpander extends Function<Query, List<Query>> {
```

Comment:
i'm wondering if this is something that should be used inside `RetrievalAugmentationAdvisor` or it is used in code at higher level.

Reply:
I haven't adopted that in `RetrievalAugmentationAdvisor` yet, but I'll do so in a next PR where I introduce the Router and Fuser components that are needed to handle multiple queries/retrievals.